src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -3,52 +3,64 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryCast;
-using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
 using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
 using Roslyn.Test.Utilities;
 using Xunit;
+using VerifyCS = Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions.CSharpCodeFixVerifier<
+    Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryCast.CSharpRemoveUnnecessaryCastDiagnosticAnalyzer,
+    Microsoft.CodeAnalysis.CSharp.RemoveUnnecessaryCast.CSharpRemoveUnnecessaryCastCodeFixProvider>;
 
 namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.RemoveUnnecessaryCast
 {
-    public partial class RemoveUnnecessaryCastTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    public class RemoveUnnecessaryCastTests
     {
-        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
-            => (new CSharpRemoveUnnecessaryCastDiagnosticAnalyzer(), new CSharpRemoveUnnecessaryCastCodeFixProvider());
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
+        public void TestStandardProperties()
+            => VerifyCS.VerifyStandardProperties();
 
         [WorkItem(545979, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/545979")]
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
         public async Task DontRemoveCastToErrorType()
         {
-            await TestMissingInRegularAndScriptAsync(
+            var source =
 @"class Program
 {
     static void Main()
     {
         object s = "";
-        foreach (object x in [|(ErrorType)s|])
+        foreach (object x in ({|CS0246:ErrorType|})s)
         {
         }
     }
-}");
+}";
+
+            await VerifyCS.VerifyCodeFixAsync(
+                source,
+                new[]
+                {
+                    // /0/Test0.cs(5,20): error CS1010: Newline in constant
+                    DiagnosticResult.CompilerError("CS1010").WithSpan(5, 20, 5, 20),
+                    // /0/Test0.cs(5,22): error CS1002: ; expected
+                    DiagnosticResult.CompilerError("CS1002").WithSpan(5, 22, 5, 22),
```

Comment:
📝 Compiler errors are written in full-form for cases where I believe the compiler error is accidental. In a follow-up, these should likely either be corrected in the source of the test itself, or converted to markup syntax if the error is intentional.